types/prosemirror-model/index.d.ts

```diff
@@ -556,7 +571,12 @@ declare class ProsemirrorNode<S extends Schema = any> {
      * block node is started. When `leafText` is given, it'll be
      * inserted for every non-text leaf node encountered.
      */
-    textBetween(from: number, to: number, blockSeparator?: string, leafText?: string): string;
+    textBetween(
+        from: number,
+        to: number,
+        blockSeparator?: string | null,
+        leafText?: string | null | ((node: ProsemirrorNode) => string),
```

Comment:
The [official documentation](https://prosemirror.net/docs/ref/#model.Node.descendants) doesn't contain the fourth parameter (`index`), hence it is not part of the public API and should not be included in the type definitions. If you believe the official documentation to be wrong, please send a PR to [ProseMirror/prosemirror-model](https://github.com/).

Reply:
This `leafText` is part of the documentation: https://prosemirror.net/docs/ref/#model.Fragment.textBetween
<img width="1078" alt="image" src="https://user-images.githubusercontent.com/3452187/143170993-2eeb78d6-6c00-4863-a134-cd72ff14cf39.png">

Reply:
@RatoX It looks like @timjbs's comment was added both (here)[https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57313#discussion_r755472555] and (here)[https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57313#discussion_r755474316]. I think this was unintentional and that the first comment is the relevant one.